src/__tests__/rerender.test.js

```diff
@@ -1,41 +1,50 @@
-/**
- * @jest-environment jsdom
- */
+import { VERSION as SVELTE_VERSION } from 'svelte/compiler'
 import { describe, expect, test, vi } from 'vitest'
-import { writable } from 'svelte/store'
 
-import { act, render, waitFor } from '..'
-import Comp from './fixtures/Rerender.svelte'
+import { act, render, screen } from '..'
+import Comp from './fixtures/Comp.svelte'
 
-test('mounts new component successfully', async () => {
-  const onMounted = vi.fn()
-  const onDestroyed = vi.fn()
+describe('rerender', () => {
+  test('updates props', async () => {
+    const { rerender } = render(Comp, { name: 'World' })
+    const element = screen.getByText('Hello World!')
 
-  const { getByTestId, rerender } = render(Comp, {
-    props: { name: 'World 1', onMounted, onDestroyed },
+    await rerender({ name: 'Dolly' })
+
+    expect(element).toHaveTextContent('Hello Dolly!')
   })
 
-  const expectToRender = (content) =>
-    waitFor(() => {
-      expect(getByTestId('test')).toHaveTextContent(content)
-      expect(onMounted).toHaveBeenCalledOnce()
-    })
+  test('warns if incorrect arguments shape used', async () => {
+    vi.stubGlobal('console', { warn: vi.fn() })
+
+    const { rerender } = render(Comp, { name: 'World' })
+    const element = screen.getByText('Hello World!')
 
-  await expectToRender('Hello World 1!')
+    await rerender({ props: { name: 'Dolly' } })
 
-  console.warn = vi.fn()
+    expect(element).toHaveTextContent('Hello Dolly!')
+    expect(console.warn).toHaveBeenCalledOnce()
+    expect(console.warn).toHaveBeenCalledWith(
+      expect.stringMatching(/deprecated/iu)
+    )
+  })
 
-  rerender({ props: { name: 'World 2' } })
-  await expectToRender('Hello World 2!')
-  expect(onDestroyed).not.toHaveBeenCalled()
+  test('change props with accessors', async () => {
```

Comment:
This test moved over from `render.test.js` because it has more in common with `rerender.test.js`